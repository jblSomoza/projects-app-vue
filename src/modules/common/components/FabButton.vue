<template>
  <button
    :class="['btn btn-circle btn-secondary fixed bottom-10 right-10', position]"
    @click="$emit('click')"
  >
    <slot></slot>
  </button>
</template>

<script lang="ts" setup>
interface Props {
  position?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
}

withDefaults(defineProps<Props>(), {
  position: 'bottom-right',
});

defineEmits(['click']);
</script>

<style scoped>
.top-left {
  top: 10px;
  left: 10px;
}

.top-right {
  top: 10px;
  right: 10px;
}

.bottom-left {
  bottom: 10px;
  left: 10px;
}

.bottom-right {
  bottom: 10px;
  right: 10px;
}
</style>
