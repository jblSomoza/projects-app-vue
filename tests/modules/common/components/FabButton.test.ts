import { shallowMount } from '@vue/test-utils';
import FabButton from '../../../../src/modules/common/components/FabButton.vue';

describe('FabButton', () => {
    test('renders with default position', async () => {
        const wrapper = shallowMount(FabButton);
        const buttonClases = wrapper.find('button').classes();
        const classesToHave = buttonClases;

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.props().position).toBe('bottom-right');
        expect(buttonClases).toEqual(expect.arrayContaining(classesToHave));
    });

    test('renders with custom position', async () => {
        const wrapper = shallowMount(FabButton, {
            props: {
                position: 'top-left'
            }
        });
        const buttonClases = wrapper.find('button').classes();
        const classesToHave = buttonClases;

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.props().position).toBe('top-left');
        expect(buttonClases).toEqual(expect.arrayContaining(classesToHave));
    });

    test('renders slot content inside button', async () => {
        const wrapper = shallowMount(FabButton, {
            slots: {
                default: '<div>Test</div>'
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find('button').text()).toBe('Test');
        expect(wrapper.find('button').find('div').exists()).toBe(true);
    });
});