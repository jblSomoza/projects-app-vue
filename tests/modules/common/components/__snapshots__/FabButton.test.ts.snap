// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FabButton > renders slot content inside button 1`] = `
"<button data-v-a0e39dc5="" class="btn btn-circle btn-secondary fixed bottom-10 right-10 bottom-right">
  <div>Test</div>
</button>"
`;

exports[`FabButton > renders with custom position 1`] = `"<button data-v-a0e39dc5="" class="btn btn-circle btn-secondary fixed bottom-10 right-10 top-left"></button>"`;

exports[`FabButton > renders with default position 1`] = `"<button data-v-a0e39dc5="" class="btn btn-circle btn-secondary fixed bottom-10 right-10 bottom-right"></button>"`;
